File: tkpddiscovery/src/main/java/com/tokopedia/discovery/util/NpaLinearLayoutManager.java
package com.tokopedia.discovery.util;

import android.content.Context;
import androidx.recyclerview.widget.RecyclerView;

import com.tkpd.library.ui.view.LinearLayoutManager;

/**
 * Created by nakama on 09/12/16.
 */

public class NpaLinearLayoutManager extends LinearLayoutManager {

    public NpaLinearLayoutManager(Context context) {
        super(context);
    }

    public NpaLinearLayoutManager(Context context, int orientation, boolean reverseLayout) {
        super(context, orientation, reverseLayout);
    }

    public NpaLinearLayoutManager(RecyclerView view) {
        super(view);
    }

    public NpaLinearLayoutManager(RecyclerView view, int orientation, boolean reverseLayout) {
        super(view, orientation, reverseLayout);
    }

    @Override
    public boolean supportsPredictiveItemAnimations() {
        return false;
    }
}

File: features/content/feed/src/main/java/com/tokopedia/feedplus/view/util/NpaLinearLayoutManager.kt
package com.tokopedia.feedplus.view.util

import android.content.Context
import android.util.AttributeSet
import androidx.recyclerview.widget.LinearLayoutManager

/**
 * @author by astidhiyaa on 30/08/22
 */
class NpaLinearLayoutManager : LinearLayoutManager {
    /**
     * Disable predictive animations. There is a bug in RecyclerView which causes views that
     * are being reloaded to pull invalid ViewHolders from the internal recycler stack if the
     * adapter size has decreased since the ViewHolder was recycled.
     */
    override fun supportsPredictiveItemAnimations(): Boolean {
        return false
    }

    constructor(context: Context?) : super(context) {}
    constructor(context: Context?, orientation: Int, reverseLayout: Boolean) : super(
        context,
        orientation,
        reverseLayout
    ) {
    }

    constructor(
        context: Context?,
        attrs: AttributeSet?,
        defStyleAttr: Int,
        defStyleRes: Int
    ) : super(context, attrs, defStyleAttr, defStyleRes) {
    }
}
